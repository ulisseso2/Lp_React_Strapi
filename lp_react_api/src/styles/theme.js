export const theme = {
  colors: {
    primary: '#457B9D',
    light: '#A8DADC',
    dark: '#1D3557',
    hot: '#E63946',
    white: '#f7fff7',
    black: '#343434',
  },
  fonts: {
    family: {
      main: "'Open Sans', sans-serif",
      title: "'Montserrat', sans-serif",
      body: "'Open Sans', sans-serif",
    },
  },
};
//tema padrão da aplicação usado em vários componentes
