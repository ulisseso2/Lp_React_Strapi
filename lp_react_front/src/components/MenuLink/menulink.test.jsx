import { screen } from '@testing-library/react';
import { MenuLink } from '.';
import { renderTheme } from '../../styles/render.theme';
describe('<MenuLink>', () => {
  it('should a link', () => {
    renderTheme(<MenuLink link={'http://localhost'}>Children</MenuLink>);
    expect(screen.getByRole('link', { nome: 'Children' })).toBeInTheDocument();
    expect(screen.getByRole('link', { name: 'Children' })).toHaveAttribute(
      'target',
      '_self',
    );
  });
  it('should a link', () => {
    renderTheme(
      <MenuLink link={'http://localhost'} newTab={true}>
        Children
      </MenuLink>,
    );

    expect(screen.getByRole('link', { name: 'Children' })).toHaveAttribute(
      'target',
      '_blank',
    );
  });
  it('should match snapshot', () => {
    const { container } = renderTheme(
      <MenuLink link={'http://localhost'}>Children</MenuLink>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
