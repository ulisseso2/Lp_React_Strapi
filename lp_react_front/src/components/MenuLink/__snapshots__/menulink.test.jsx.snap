// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink> should match snapshot 1`] = `
<a
  class="sc-bczRLJ evjXeG"
  href="http://localhost"
  target="_self"
>
  Children
</a>
`;
